src/job_definition.jl

```diff
@@ -131,36 +126,52 @@ end
 
 Deregisters an AWS Batch job.
 """
-function deregister(definition::JobDefinition)
+function deregister(definition::JobDefinition; aws_config::AbstractAWSConfig=global_aws_config())
     debug(logger, "Deregistering job definition \"$(definition.arn)\"")
-    resp = deregister_job_definition(Dict("jobDefinition" => definition.arn))
+    resp = @mock Batch.deregister_job_definition(definition.arn; aws_config=aws_config)
     info(logger, "Deregistered job definition \"$(definition.arn)\"")
 end
 
 """
-    isregistered(definition::JobDefinition) -> Bool
+    isregistered(definition::JobDefinition; aws_config=global_aws_config()) -> Bool
 
 Checks if a JobDefinition is registered.
 """
-function isregistered(definition::JobDefinition)
-    j = describe(definition)
+function isregistered(definition::JobDefinition; aws_config::AbstractAWSConfig=global_aws_config())
+    j = describe(definition; aws_config)
     return any(d -> d["status"] == "ACTIVE", get(j, "jobDefinitions", []))
 end
 
 """
-    describe(definition::JobDefinition) -> Dict
+    list_job_definitions(;aws_config=global_aws_config())
+
+Get a list of `JobDefinition` objects via `Batch.decsribe_job_definitions()`.
+"""
+function list_job_definitions(;aws_config::AbstractAWSConfig=global_aws_config())
+    [JobDefinition(a["jobDefinitionArn"]) for
+     a ∈ Batch.describe_job_definitions(;aws_config)["jobDefinitions"]]
+end
```

Comment:
Not a big fan of nesting calls within each other, I *think* there's something in [BlueStyle](https://github.com/invenia/BlueStyle/) about it. But I feel like chaining them together just decreases readability and scalability if there was a need to do something conditional in the future.

I know Julia is more-so a scientific language, but personally really hate all the unicode scattered through out packages and would like to avoid it as much as possible. It just makes writing code more difficult needing to remember how to insert these characters (or looking for one and copying/pasting them in rather than just typing `in`.

```suggestion
function list_job_definitions(;aws_config::AbstractAWSConfig=global_aws_config())
    job_definitions = Batch.describe_job_definitions(; aws_config)["jobDefinitions"]
    
    return [JobDefintiion(jd["jobDefinitionArn"]) for jd in job_definitions]
end
```